src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -249,6 +249,7 @@
  * <ol>
  * <li>the alignment constraint of {@code S} is set to its
  *     <a href="MemoryLayout.html#layout-align">natural alignment</a>, and</li>
+ * <li>{@code S.elementLayout()} is not a padding layout.</li>
```

Comment:
I think the existing text already covers this case, so no addition should be needed. Padding layouts on their own are not supported layouts. (Note e.g. how for group layouts we say: 'each member layout ... is either a padding layout or a layout supported by {@code NL}'.

Though, either way we'll need a CSR for the behavior change.